### PR TITLE
Improve docs of broker pallet

### PR DESCRIPTION
**Pull request by eskimor**
Small adjustments which should make understanding what is going on much easier for future readers.

Initialization is a bit messy, the very least we should do is adding documentation to make it harder to use wrongly.

I was thinking about calling `request_core_count` right from `start_sales`, but as explained in the docs, this is not necessarily what you want.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5780237

File: substrate/frame/broker/src/lib.rs [hunk 1]
```diff
@@ -548,16 +548,26 @@ pub mod pallet {
 		///
 		/// - `origin`: Must be Root or pass `AdminOrigin`.
 		/// - `initial_price`: The price of Bulk Coretime in the first sale.
-		/// - `core_count`: The number of cores which can be allocated.
+		/// - `total_core_count`: This is the total number of cores the relay chain should have
+		/// after the sale concludes.
```
Comment (bkchr) [inline substrate/frame/broker/src/lib.rs:552]:
```suggestion
		/// after the sale concludes.
		///
```

**Review by bkchr — APPROVED** [review]

File: substrate/frame/broker/src/lib.rs [hunk 2]
```diff
@@ -548,16 +548,26 @@ pub mod pallet {
...
+		/// cores will just be `Idle`. If you are actually reducing the number of cores, you should
+		/// call `request_core_count`, right before the next sale, to avoid shutting down tasks too
+		/// early.
 		#[pallet::call_index(4)]
 		#[pallet::weight(T::WeightInfo::start_sales((*core_count).into()))]
 		pub fn start_sales(
 			origin: OriginFor<T>,
 			initial_price: BalanceOf<T>,
-			core_count: CoreIndex,
+			total_core_count: Option<CoreIndex>,
```
Comment (bkchr) [inline substrate/frame/broker/src/lib.rs:567]:
```suggestion
			total_core_count: CoreIndex,
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5785612

**Review by seadanda — APPROVED** [review]

File: substrate/frame/broker/src/lib.rs [hunk 3]
```diff
@@ -550,16 +550,27 @@ pub mod pallet {
...
+		/// call `request_core_count`, right before the next sale, to avoid shutting down tasks too
+		/// early.
 		#[pallet::call_index(4)]
-		#[pallet::weight(T::WeightInfo::start_sales((*core_count).into()))]
+		#[pallet::weight(T::WeightInfo::start_sales((*total_core_count).into()))]
 		pub fn start_sales(
 			origin: OriginFor<T>,
 			initial_price: BalanceOf<T>,
-			core_count: CoreIndex,
+			total_core_count: CoreIndex,
```
Comment (sandreim) [inline substrate/frame/broker/src/lib.rs:570]: Based on the comment above the parameter should be renamed:

```suggestion
			concluding_core_count: CoreIndex,
```
or 

```suggestion
			resulting_core_count: CoreIndex,
```

**Review by sandreim — APPROVED** [review]

**Review by tdimitrov — APPROVED** [review]

**Pull request merged**